libbeat/cmd/instance/beat.go

```diff
@@ -855,6 +862,30 @@ func (b *Beat) loadDashboards(ctx context.Context, force bool) error {
 	return nil
 }
 
+// warnAboutElasticsearchVersion registers a global callback to warn users once
+// if they try to connect to an older Elasticsearch version than the Beat version.
+func (b *Beat) warnAboutElasticsearchVersion() {
+	if b.Config.Output.Name() != "elasticsearch" {
+		return
+	}
+
+	elasticsearch.RegisterGlobalCallback(func(conn *eslegclient.Connection) error {
+		warnAboutOldES.Do(func() {
+			esVersion := conn.GetVersion()
+			beatVersion, err := common.NewVersion(b.Info.Version)
+			if err != nil {
+				logp.Debug("beat", "Failed to determine Beat version to compare it with ES version: %+v", err)
+				return
+			}
+
+			if esVersion.LessThan(beatVersion) {
+				logp.Warn("Connecting to older version of Elasticsearch. From 8.1, it is going to be disabled by default.")
```

Comment:
```suggestion
				logp.Warn("Connecting to older version of Elasticsearch. From 8.1, connecting to older versions will be disabled by default.")
```
I'd prefer this to be a little be more specific :)